### DISCUSSION TITLE
Guest::ShouldGoOnRide logic when guest already “heading for ride”

### DISCUSSION THREAD

**Discussion by reticulatingsplines** [OpenRCT2/OpenRCT2#15903] [Q&A]
Seems like I could just be totally missing something, but the comment below in the ShouldGoOnRide function in Guest.cpp doesn’t seem to reflect what the code actually does. It would appear that, if the guest is already heading for a ride, they indeed do this “basic” intensity check, and return false if the ride is too generically intense, but it’s not like there’s some early return true line…it just goes on to the weather check anyways, which isn’t nested inside of the if function or anything. Is this an oversight or does the comment not actually reflect the intended function? Or am I not seeing a bracket?

```
if (ride_has_ratings(ride))
            {
                // If a peep has already decided that they're going to go on a ride, they'll skip the weather and
                // excitement check and will only do a basic intensity check when they arrive at the ride itself.
                if (ride->id == GuestHeadingToRideId)
                {
                    if (ride->intensity > RIDE_RATING(10, 00) && !gCheatsIgnoreRideIntensity)
                    {
                        peep_ride_is_too_intense(this, ride, peepAtRide);
                        return false;
                    }
                }
                // Peeps won't go on rides that aren't sufficiently undercover while it's raining.
                // The threshold is fairly low and only requires about 10-15% of the ride to be undercover.
                if (climate_is_raining() && (ride->sheltered_eighths) < 3)
                {
                    if (peepAtRide)
                    {
                        InsertNewThought(PeepThoughtType::NotWhileRaining, ride->id);
                        if……..///and on and on, further tests
```

**Comment by reticulatingsplines** [comment]
PS, in-game behavior seems to confirm the comment isn’t true; I can play with the code to manually set a guest’s GuestHeadingToRideId value to a low-intensity ride, and they will still say “I want to go on something more thrilling than RIDE” and turn around at the queue banner.

**Comment by ZehMatt** [comment]
I'm not sure I quite follow. To me it looks correct?

**Comment by reticulatingsplines** [comment]
If ride->id == GuestHeadingToRideId, and the ride’s intensity is 3, the very next thing that happens is a weather check, no?

**Comment by ZehMatt** [comment]
At a closer look I guess the comment is incorrect to some degree.

**Comment by reticulatingsplines** [comment]
It seems to me like the intended effect would be:

                if (ride->id == GuestHeadingToRideId)
                {
                    if (ride->intensity > RIDE_RATING(10, 00) && !gCheatsIgnoreRideIntensity)
                    {
                        peep_ride_is_too_intense(this, ride, peepAtRide);
                        return false;
                    }
                    else
                    {
                        return true;
                    }
                }

**Reply by ZehMatt** [reply]
We should validate that this is actually the case in vanilla first.

**Reply by reticulatingsplines** [reply]
Yeah, I think the code I suggested is representative of the vanilla game's logic, at least here in RCT Classic. Here's some guests that just bounced off the queue entrance of Iceberg Golf, which is nice and sleep-inducing with a intensity rating of 0.92. 

![3 guests deciding "I want to go on something more thrilling than Iceberg Golf"](https://i.ibb.co/BTVT4Cq/IMG-0330.png)

Note that despite how happy they are, their preference for rides with intensity >4 means they're never going to randomly wander onto Iceberg Golf. However, here's Bruce, who's not quite as happy, but also has the same preference for rides with intensity >4. He's actively heading towards Iceberg Golf.

![Guest actively heading towards Iceberg Golf with intensity preference >4](https://i.ibb.co/SR63sds/IMG-0348.png)

He gets right on:

![Guest actively heading towards Iceberg Golf gets in queue, despite intensity preference >4](https://i.ibb.co/KxZzd3m/IMG-0349.png)

Here are some guests bouncing off of the queue for Iceberg Golf because it's raining:

![Guests deciding "I'm not going on Iceberg Golf while it's raining"](https://i.ibb.co/MRPJc9w/IMG-0350.png)

However, here's Graham, who's heading for Iceberg Golf in a 35F (sorry, 2C) March rainstorm (I guess they DID go to a theme park on an iceberg, what can you expect):

![Guest heading towards Iceberg Golf while it's raining](https://i.ibb.co/dpG192J/IMG-0334.png)

And he gets in the queue:

![Guest that was heading towards Iceberg Golf gets in the queue despite the rain](https://i.ibb.co/f2rVvx5/IMG-0335.png)

They do, however, do that "basic intensity check". Here's some guests I coaxed to head for Vertical Drop Roller Coaster 1 with a free voucher campaign (which generates guests that already have GuestHeadingToRideId set to the advertised ride) - nobody seemed interested in astronaut training if they had to pay for it. 

![Guests heading for Vertical Drop Roller Coaster 1, which has an intensity of 18.27](https://i.ibb.co/LRrch7W/IMG-0344.png)

They still have an overriding sense of self-preservation:

![Guests who were heading for Vertical Drop Roller Coaster 1 bounce off the queue entry, thinking "Vertical Drop Roller Coaster 1 looks too intense for me"](https://i.ibb.co/VNGvbSK/IMG-0346.jpg)

**Reply by reticulatingsplines** [reply]
Just wanted to add that I also confirmed this identical behavior in RCT1. Guests "heading to ride" ignore rain and ignore intensity being outside of acceptable range regardless of happiness.

**Reply by reticulatingsplines** [reply]
(They do, however, still do the checks outside of the `if (ride_has_ratings(ride))` block - they will likely bounce off if stats aren't calculated when they arrive and they won't pay 20 regardless of value)
